onnxruntime/core/mlas/lib/kleidiai/sgemm_kleidiai.cpp

```diff
@@ -14,6 +14,16 @@
 #include "kai/ukernels/matmul/pack/kai_rhs_pack_nxk_f32p2vlx1biasf32_f32_f32_sme.h"
 #include "mlasi_kleidiai.h"
 
+
+// Thread-local reusable buffers to reduce allocation overhead across tiles.
+struct KaiTlsBuffers {
+    std::vector<float> output_tile;
+    std::vector<float> bias_zero;
+    std::vector<std::byte> rhs_packed;
+    std::vector<std::byte> lhs_packed;
+};
+static thread_local KaiTlsBuffers g_kai_tls;
```

Comment:
Please note that ORT as a library could not control when a thread_local variable is deallocated.  As the destructor of this var does not depend on anything except the C++ runtime, I feel it's fine.